### PR TITLE
Add --hard flag to clean and update to go1.18

### PR DESCRIPTION
**Pull request by holisticode**
Closes https://github.com/ava-labs/avalanche-cli/issues/103 and https://github.com/ava-labs/avalanche-cli/issues/101

### CODE REVIEW

**Review by felipemadero — COMMENTED** [review]
probably you need to bump go version also in CI actions

File: cmd/networkcmd/clean.go [hunk 1]
```diff
@@ -34,4 +49,17 @@ func clean(cmd *cobra.Command, args []string) {
...
+		binDir := filepath.Join(app.GetBaseDir(), constants.AvalancheCliBinDir)
+		cleanBins(binDir)
+	}
+}
+
+func cleanBins(dir string) {
+	if err := os.RemoveAll(dir); err != nil {
+		ux.Logger.PrintToUser("Removal failed: %s", err)
+	}
+	ux.Logger.PrintToUser("All existing binaries removed. To continue using the tool, install latest binaries again.")
```
Comment (felipemadero) [inline cmd/networkcmd/clean.go:64]: Confused by the last sentence. The user may think he needs to install binaries by itself, but really the tool is going to make that for him on the next command execution. Probably just drop the last sentence?

Comment (holisticode) [reply cmd/networkcmd/clean.go:64]: After the fact I agree

**Review by felipemadero — COMMENTED** [review]
you may also want to bump go in .golangci.yml

**Review by felipemadero — APPROVED** [review]

**Pull request merged**